### DISCUSSION TITLE
throttle과 debounce 당신의 선택은?

### DISCUSSION THREAD

**Discussion by byhhh2** [woowacourse-study/2022-code-review-study#50] [Youtube] [answered]
throttle과 debounce 당신의 선택은?

무한 스크롤의 문제 (스크롤 이벤트가 발생될 때마다 API를 요청)를 해결하기 위해 throttle과 debounce 두가지 방식이 추천되었는데요. 어떤 방법이 이번 미션에 더 적절할까요?

throttle : 일정 시간동안 이벤트가 한번만 발생하도록 한다.
debounce : 여러 이벤트가 발생했을 때 마지막 이벤트만 실행되도록 한다.

참고할 [PR](https://github.com/woowacourse/javascript-youtube-classroom/pull/107#discussion_r825321999)

**Comment by hwangstar156** [comment, marked as answer]
저는 throttle을 추천합니당. 

디바운싱(trailing edge)은 ajax 요청을 한꺼번에 모아서 마지막에 요청을 한번 보내는것으로 효율적인 api요청을 위한것입니당.
이렇게 이용하면 스크롤을 끝을 찍고 다시 올리면 로딩이 진행되지 않게 됩니다. 스크롤이 아직 끝나지 않았기때문이죠

쓰로틀은 반드시 일정시간내에 한번의 이벤트 실행만큼은 보장하기에 쓰로틀이 좋을거 같습니당

**Reply by hwangstar156** [reply]
냠

**Reply by wonsss** [reply]
내공 킬러다
